File: Test/AdventureWorksFunctionalModel/Person/AddressType.cs
namespace AW.Types;

[Bounded]
public class AddressType : IHasModifiedDate, IHasRowGuid
{
    [Hidden]
    public int AddressTypeID { get; init; }

    [Hidden]
    public string Name { get; init; } = "";


    [Hidden]
    [Versioned]
    public DateTime ModifiedDate { get; init; }

    [Hidden]
    public Guid rowguid { get; init; }

    public override string ToString() => Name;
}

File: Test/AdventureWorksLegacyModelVB/Person/AddressType.vb
Namespace AW.Types

    '<Bounded>
    Partial Public Class AddressType

        Public Property AddressTypeID() As Integer

#Region "Name"
        Public mappedName As String
        Friend myName As TextString

        '<MemberOrder(1)>
        Private ReadOnly Property Name As TextString
            Get
                Return If(myName, New TextString(mappedName, Function(v) mappedName = v))
            End Get
        End Property

        Public Sub AboutName(a As FieldAbout, Name As TextString)
            Select Case a.TypeCode
                Case AboutTypeCodes.Visible
                    a.Visible = True
            End Select
        End Sub
#End Region

#Region "ModifiedDate"
        Public mappedModifiedDate As Date
        Friend myModifiedDate As TimeStamp

        '<MemberOrder(99)>
        Private ReadOnly Property ModifiedDate As TimeStamp
            Get
                Return If(myModifiedDate, New TimeStamp(mappedModifiedDate, Function(v) mappedModifiedDate = v))
            End Get
        End Property

        Public Sub AboutModifiedDate(a As FieldAbout, ModifiedDate As TimeStamp)
            Select Case a.TypeCode
                Case AboutTypeCodes.Usable
                    a.Usable = False
            End Select
        End Sub
#End Region

        Public Property RowGuid() As Guid

        Public Function Title() As Title
            Return New Title(Name)
        End Function
    End Class
End Namespace